webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/configuration/DirectiveManagement.scala

```diff
@@ -69,13 +69,24 @@ import net.liftweb.http.js.*
 import net.liftweb.http.js.JE.*
 import net.liftweb.http.js.JE.JsArray
 import net.liftweb.http.js.JsCmds.*
-import net.liftweb.json
 import net.liftweb.util.Helpers.*
 import net.liftweb.util.Helpers.TimeSpan
+import org.apache.commons.lang3.StringEscapeUtils
 import org.joda.time.DateTime
+import scala.annotation.nowarn
 import scala.xml.*
+import zio.json.*
 
-final case class JsonDirectiveRId(directiveId: String, rev: Option[String])
+final case class JsonDirectiveRId(directiveId: String, rev: Option[String]) {
+  def escaped: JsonDirectiveRId = {
+    JsonDirectiveRId(StringEscapeUtils.escapeHtml4(directiveId), rev.map(StringEscapeUtils.escapeHtml4)): @nowarn(
```

Comment:
And a last comment: I would prefer the escaping to be done closer to the actual use of the variable (as the caller is not expected to know in which context it would be used).